### PR TITLE
 mm 332 - Use SiteURL for all calls from the webapp

### PR DESCRIPTION
**Pull request by sibasankar-demansol**
Added Site Url for call from webapp

ticket here
https://github.com/mattermost/mattermost-plugin-github/issues/332

### CODE REVIEW

File: webapp/src/index.js [hunk 1]
```diff
@@ -64,3 +64,14 @@ class PluginClass {
...
+    const config = getConfig(state);
+    let basePath = '';
+    if (config && config.SiteURL) {
+        basePath = new URL(config.SiteURL).pathname;
+        if (basePath && basePath[basePath.length - 1] === '/') {
+            basePath = basePath.substr(0, basePath.length - 1);
+        }
+    }
+    return basePath;
+};
```
Comment (hanzei) [inline webapp/src/index.js:77]: Nit pick: add line break

**Review by hanzei — CHANGES_REQUESTED** [review]
The changes look good to me. :+1: 

Would you mind removing the unrelated changes like removing line breaks to simplify the diff and keep the code consistent?

File: webapp/src/index.js (hunk 1 above)
Comment (sibasankar-demansol) [reply webapp/src/index.js:83]: @hanzei sure , have resolved the comments 
hope this will work fine

Comment (hanzei) [reply webapp/src/index.js:83]: The line break is still mising.

**Review by hanzei — APPROVED** [review]
Thanks @sibasankar-demansol :+1:

File: webapp/src/index.js (hunk 1 above)
Comment (sibasankar-demansol) [reply webapp/src/index.js:83]: do we want to have a line break after return statement and closing bracket of function ?

Comment (mickmister) [reply webapp/src/index.js:83]: The line break should be before the return statement

Comment (sibasankar-umpteens) [reply webapp/src/index.js:83]: Screenshot 2021-07-15 at 10 30 07 AM


i guess the new change is not getting reflected , have attached a screenshot of same

Comment (hanzei) [reply webapp/src/index.js:83]: The UI also shows that a linebreak is missing.

Comment (sibasankar-demansol) [reply webapp/src/index.js:83]: Don't know why it is not reflecting here 
can check this 
https://github.com/mattermost/mattermost-plugin-github/blob/b392c16ddb9ff65b2cef134a798a96581802095c/webapp/src/index.js#L81

Comment (hanzei) [reply webapp/src/index.js:83]: The file you linked also doesn't have a line break in the last line. See https://raw.githubusercontent.com/mattermost/mattermost-plugin-github/b392c16ddb9ff65b2cef134a798a96581802095c/webapp/src/index.js.

Comment (sibasankar-demansol) [reply webapp/src/index.js:83]: you mean there should be a line break before and after return statement ? or line break at last line of the file

Comment (mickmister) [reply webapp/src/index.js:83]: There should be a line break at the last line of the file. Also, do you mind moving  this `getServerRoute` function to the `selectors.js` file?

**Review by mickmister — CHANGES_REQUESTED** [review]

File: webapp/src/index.js (hunk 1 above)
Comment (sibasankar-demansol) [reply webapp/src/index.js]: done

**Comment by maisnamraju** [comment]
@mickmister  looks this the changes are updated now. Please take a look, ill keep an eye if you need any changes

**Review by mickmister — APPROVED** [review]

**Review by hanzei — APPROVED** [review]

**Pull request merged**